### ISSUE TITLE
Disable warning "Not configured to insert distributed trace headers"

### ISSUE THREAD

**Issue by fabioxgn** [newrelic/newrelic-ruby-agent#341]
I'm using the default New Relic configuration and don't use distributed tracing, but after upgrading this gem this warning appears a lot on my logs, how can I disable it?

**Comment by tannalynn** [comment]
Hello @fabioxgn!
So it looks like this warning appears typically because you don't have distributed tracing enabled, but it is attempting to do something that needs distributed tracing. I assume you are not calling our api methods that deal with distributed tracing, so it does seem odd you would be getting these warnings at all.
I don't get these warnings when I run the agent in an environment with distributed tracing disabled, so I would suggest reaching out to our support team at support.newrelic.com to work with them to make sure that your distributed tracing settings are correct.
Thank you!

**Comment by fabioxgn** [comment]
Hi @tannalynn I'm using the default configuration generated by this gem, and the default is `enabled: false` ain't it? https://docs.newrelic.com/docs/agents/ruby-agent/configuration/ruby-agent-configuration#distributed_tracing

And I don't have a Pro subscription (Just infrastructure) I tried to create the ticket but it redirects me to the community, which I already asked: https://discuss.newrelic.com/t/not-configured-to-insert-distributed-trace-headers-warning/104182

Maybe I'm getting this warning because my APM subscription is "Lite"?

**Comment by tannalynn** [comment]
hi @fabioxgn 
What version of the agent are you running?
Also, are you using sidekiq?
Thanks!

**Comment by benhart** [comment]
Hi @tannalynn  
Our logs are also flooded with these messages, multiple times per second. I've tried explicitly disabling distributed tracing, with no success. 

We're using default configuration, puma on web, and sidekiq. This only started with v 6.11.0.365 of the ruby agent. 

Please advise how I can stop the logging.

Thanks!

**Comment by benhart** [comment]
Hi @tannalynn 
We dropped that in an initializer and working as expected. Thanks for that!

**Comment by henrik** [comment]
To save the next person the trouble – the current version as of that initializer was 6.11.0.365. I suggest adding something like this to the top of the initializer so you don't forget to remove it:

```
raise "Remove this fix" if defined?(NewRelic) && Gem::Version.new(NewRelic::VERSION::STRING) > Gem::Version.new("6.11.0.365")
```

The `defined?` check is needed if you don't include New Relic in every environment.

**Comment by mwlang** [comment]
@henrik I really like this idea.  I plan to start adding a preamble like this to all future patches we offer to test fixes before rolling into a release.

**Comment by tannalynn** [comment]
@fabioxgn hello! We've released version 6.12 of the agent which has this issue fixed. Thank you for bringing it to our attention!
